Exclude Google Site Kit files from JavaScript minification to fix JavaScript errors

vmanthos opened the pull request:
## Description

I've added the following:
```
$excluded_files[] = '/google-site-kit/dist/assets/js/googlesitekit-modules-(.*).js';
```
in the `get_excluded_files()` method.

Fixes #4656 

## Type of change

Please delete options that are not relevant.

- [x] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

This wasn't groomed.

## How Has This Been Tested?

I've tested this locally by installing Google Site Kit, and checking that the JavaScript errors were resolved.

# Checklist:

Please delete the options that are not relevant.

- [x My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

vmanthos committed:
Exclude Google Site Kit files

Mai-Saad commented:
@vmanthos Can see it working fine with Minify JS, Defer and Delay JS. But can see this error when Combine JS
![Screenshot from 2022-01-18 18-36-24](https://user-images.githubusercontent.com/76941962/149980580-f4503785-bc1d-4b82-83c0-46f6e5ebdd7f.png)
Can you please check that?

aaemnnosttv commented on inc/Engine/Optimization/Minify/JS/AbstractJSOptimization.php:
```diff
@@ -53,6 +53,7 @@ protected function get_excluded_files() {
 		$excluded_files[] = '/interactive-3d-flipbook-powered-physics-engine/assets/js/pdf.min.js';
 		$excluded_files[] = '/interactive-3d-flipbook-powered-physics-engine/assets/js/three.min.js';
 		$excluded_files[] = '/interactive-3d-flipbook-powered-physics-engine/assets/js/3d-flip-book.min.js';
+		$excluded_files[] = '/google-site-kit/dist/assets/js/googlesitekit-modules-(.*).js';
```
@Tabrisrp Can we please extend this to all of Site Kit's JS files? We don't load any on the front end for non-logged-in users. The pattern here is too specific as well as it would not match our runtime. See the list of all the current assets here: https://plugins.svn.wordpress.org/google-site-kit/tags/1.49.0/dist/assets/js/

I haven't tested this, but this should work assuming regular expressions are supported 🙂 

```suggestion
		$excluded_files[] = '/google-site-kit/dist/.*\.js';
```

vmanthos replied:
I've moved on with excluding all JavaScript files.

vmanthos commented:
@Mai-Saad I'm not sure the specific error you saw is related to the exclusions this PR adds. 

However, there are JavaScript errors coming from it when  **Combine JavaScript files** is active.
On my test site, the errors are specific to Google Site Kit:

![](https://i.vgy.me/cHQZxH.jpg)

![](https://i.vgy.me/KRaJGm.jpg)

The following exclusions resolved the specific errors:
```
/google-site-kit/dist/assets/js/googlesitekit-data-(.*).js
/google-site-kit/dist/assets/js/googlesitekit-widgets-(.*).js
```
I'll add these as well to the PR so you can test them.

vmanthos committed:
Add exclusions to resolve JavaScript errors when combining JS

aaemnnosttv commented:
@vmanthos please see my comment on your PR for a suggestion that should match all our JS files 🙂  https://github.com/wp-media/wp-rocket/pull/4657/files#r790655067

vmanthos commented:
@aaemnnosttv We usually exclude the files that are involved in generating errors/issues.

Let's see what our product team has to say about your request.

@piotrbak What do you think about excluding the whole list of Google Site Kit's files from JavaScript minification/combination?

piotrbak commented:
@vmanthos Yes, I think we can exclude those, since they're only for logged-in users. By doing so, we'll be protected from potential errors in the future. Let's see if we won't have any errors during the QA.

vmanthos committed:
Exclude all Google Site Kit files from JavaScript minification/combination

Following Piotr's comment (https://github.com/wp-media/wp-rocket/pull/4657#issuecomment-1020280216) I'm removing the previous exclusions and excluding all GSK's JavaScript files.

Mai-Saad reviewed (approved):
Working fine with Site Kit by Google 1.49.0 and 1.49.1
[testrail-report-346.pdf](https://github.com/wp-media/wp-rocket/files/7980521/testrail-report-346.pdf)
